Test1/ConcurrentPriorityQueue.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Threading;
+
+namespace Test1;
+
+public class ConcurrentPriorityQueue<TP, TV> : IConcurrentPriorityQueue<TP, TV> where TP : IComparable<TP>
```

Comment:
```suggestion
public class ConcurrentPriorityQueue<TPriority, TValue> : IConcurrentPriorityQueue<TPriority, TValue> where TPriority : IComparable
```
Имена типовых переменных - это имена. Поэтому они должны быть содержательными

Reply:
Извините, мы из котика, там так не принято: https://github.com/JetBrains/kotlin/blob/f3385fbdcb8b9dc29661a8a4973c855cdcf73767/core/builtins/native/kotlin/Collections.kt#L295

Reply:
-0.5